compiler/annotation/type_seeder.rs

```diff
@@ -317,6 +320,9 @@ impl<'this, Snapshot: ReadableSnapshot> TypeGraphSeedingContext<'this, Snapshot>
                     any |= self.annotate_some_unannotated_vertex(nested_graph, context)?;
                 }
             }
+            for optional in &mut graph.nested_optionals {
+                any |= self.annotate_some_unannotated_vertex(optional, context)?;
+            }
```

Comment:
This should not be here. 
Optionals cannot affect the parent's annotations, and should be handled as negations are in `seed_types_in_nested_negations_and_optionals`.